wrapanapi/systems/msazure.py

```diff
@@ -669,6 +677,23 @@ def list_free_nics(self, nic_template, resource_group=None):
         ips = self.network_client.network_interfaces.list(resource_group_name=resource_group)
         return [ip.name for ip in ips if not ip.virtual_machine and nic_template in ip.name]
 
+    def list_all_free_nics(self, resource_group=None):
```

Comment:
I don't think this should be a separate method from `list_free_nics`, its behavior is _very_ similar. I think L678 can be modified (probably expanded into full for loop) in order to handle `nic_template=None` default, only taking it into account when it has been passed. This will keep the same function args/kwargs and not change anything for currently callers of the function, but provide an additional mode of operation if called without a `nic_template`.